apps/server/src/modules/tool/school-external-tool/service/school-external-tool.service.spec.ts

```diff
@@ -28,19 +33,32 @@ describe('SchoolExternalToolService', () => {
 					provide: ExternalToolService,
 					useValue: createMock<ExternalToolService>(),
 				},
+				{
+					provide: SchoolExternalToolValidationService,
+					useValue: createMock<SchoolExternalToolValidationService>(),
+				},
+				{
+					provide: ToolFeatures,
+					useValue: {
+						toolStatusWithoutVersions: false,
+					},
+				},
 			],
 		}).compile();
 
 		service = module.get(SchoolExternalToolService);
 		schoolExternalToolRepo = module.get(SchoolExternalToolRepo);
 		externalToolService = module.get(ExternalToolService);
+		schoolExternalToolValidationService = module.get(SchoolExternalToolValidationService);
+		toolFearures = module.get(ToolFeatures);
 	});
 
-	const setup = () => {
+	const legacySetup = () => {
```

Comment:
Why? Please follow test structure and move the setup to an own describe so that you can use the name "setup" for both. If you want to extract things from setups then use a more describing name like: mockExternalTools and use it in the setup functions

Reply:
i wanted to prevent refactoring the whole file. Because i would need to pull the setup funktion into every describe of existing tests to be able to use it in my test...

Reply:
Please add this to 1337 with removing the feature flag and so on